### PR TITLE
Refactor accessibility map polygon generation to use postgis functions

### PR DESCRIPTION
**Pull request by greenscientist**
Our accessibility map generate was taking a log of time. They were using turf function in javascript which are not the best performing options.

This change the code to store the result in a temporary postgis table and merge the circle via St_Union.

We also took the opportunity to reduce the circule precision and increase performance by lowering the quad_segs parameter. A full analysis is available in comment in the code.

With these changes, we go from about 18s for a polygon generation to less than 3s. (For a point near Polytechnique Montreal, with an STM GTFS.)

The unit test we redone with Claude.

## Summary by CodeRabbit

- Refactor
  - Polygon generation now uses a database-backed path for unioning/clipping circles; progress reporting preserved.
- New Features
  - Added a DB-backed polygon clipping utility that returns MultiPolygon results from circle inputs.
- Bug Fixes
  - Better handling of empty/single-node results, deterministic temp-table behavior, and clearer error handling when polygon generation fails.
- Tests
  - Expanded unit and integration tests covering DB-backed clipping, multi/single-polygon cases, edge conditions, concurrency, and error paths.
- Chores
  - Removed an unused polygon-clipping dependency.

### CODE REVIEW

File: packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        const startTime = Date.now();
+        const result = await clipPolygon(circles);
+        const duration = Date.now() - startTime;
+
+        expect(Array.isArray(result)).toBe(true);
+        expect(result.length).toBeGreaterThan(0);
+
+        // Should complete in reasonable time (< 5 seconds for 50 circles)
+        expect(duration).toBeLessThan(5000);
+
```
[the remark this answers is not in the document]
Comment (tahini) [reply packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts]: @coderabbitai has a point. anyway, our tests have a 5 or 10 seconds timeout so should fail if the timing is unreasonable.

File: packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+            AND tablename LIKE 'temp_circles_%'
+        `);
+
+        // Should be 0 or very low if other tests are running in parallel
+        expect(parseInt(tempTables.rows[0].count)).toBeLessThan(10);
+    });
+
+    test('should throw error when database connection fails', async () => {
+        // This test ensures error handling works
+        // We can't easily simulate DB failure, but we can test the error path exists
```
Comment (tahini) [inline packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts]: This does not seem to test error failure at all. You can probably send an invalid value as parameter, with a cast as any and make sure it throws an error (if it's the case)

File: packages/transition-backend/src/models/db/geometryUtils.db.queries.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+            circles.forEach((circle) => {
+                valuePlaceholders.push('(ST_SetSRID(ST_MakePoint(?, ?), 4326)::geography, ?)');
+                bindings.push(circle.center[0], circle.center[1], circle.radiusKm * 1000);
+            });
+
+            // Create temp table and generate circles in one go
+            /**
+             * Ajusting the quad_segs will have an impact on speed and precision.
+             * quad_segs is the number of segment to approximate a quarter of a circle.
```
Comment (tahini) [inline packages/transition-backend/src/models/db/geometryUtils.db.queries.ts]: ah voilà ce que je cherchais! c'est bon

File: packages/transition-backend/src/services/accessibilityMap/__tests__/TransitAccessibilityMapCalculator.test.ts [hunk 4]
```diff
@@ -13,11 +12,28 @@ import {
...
 } from 'chaire-lib-common/lib/config/osm/osmMappingDetailedCategoryToCategory';
-import polygonClipping from 'polygon-clipping';
 
 import { TransitAccessibilityMapCalculator } from '../TransitAccessibilityMapCalculator';
 import nodesDbQueries from '../../../models/db/transitNodes.db.queries';
 import placesDbQueries from '../../../models/db/places.db.queries';
+import { TransitAccessibilityMapWithPolygonResult } from 'transition-common/lib/services/accessibilityMap/TransitAccessibilityMapResult';
+
+// Mock knex for PostGIS operations
+jest.mock('chaire-lib-backend/lib/config/shared/db.config', () => {
```
Comment (tahini) [inline packages/transition-backend/src/services/accessibilityMap/__tests__/TransitAccessibilityMapCalculator.test.ts]: Ici, tu peux maintenant juste mocker geometryUtils.db.queries.ts et la fonction et lui faire retourner ce que tu veux, je crois que tu n'as plus besoin de cela.

File: packages/transition-backend/src/models/db/geometryUtils.db.queries.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+export async function clipPolygon(circles: Array<{ center: [number, number]; radiusKm: number }>): Promise<any> {
+    if (circles.length === 0) {
+        return [];
+    }
+
+    try {
+        return await knex.transaction(async (trx) => {
+            // Set higher work memory, with the default 4M, the request were swapping
+            // to disk. Given them 5 times that seems to work well and still
+            // safe when doing many requests in parallel.
```
[the remark this answers is not in the document]
Comment (greenscientist) [reply packages/transition-backend/src/models/db/geometryUtils.db.queries.ts]: It's the postgresql default.

File: packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+        // Test error handling by passing invalid data that would cause DB error
+        // For example, NaN or invalid coordinates
+        const invalidCircles = [
+            { center: [NaN, 45.5] as [number, number], radiusKm: 0.5 }
+        ];
+
+        await expect(clipPolygon(invalidCircles)).rejects.toThrow();
+    });
+
+    test('should verify quad_segs parameter produces expected point count', async () => {
```
[the remark this answers is not in the document]
Comment (greenscientist) [reply packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts]: Pas pour le moment. Un jour oui, quand on laissera au user le choix d'ajustement la précision

File: packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+        // Check that no temp tables are left behind
+        // Note: temp tables auto-drop on commit, so there should be none or very few
+        const tempTables = await knex.raw(`
+            SELECT COUNT(*) as count
+            FROM pg_tables
+            WHERE schemaname LIKE 'pg_temp%'
+            AND tablename LIKE 'temp_circles_%'
+        `);
+
+        // Should be 0 or very low if other tests are running in parallel
```
[the remark this answers is not in the document]
Comment (greenscientist) [reply packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts]: How does it ensure it? If I execute the command twice on my system, anything warant the sequential access to the DB?

**Comment by greenscientist** [comment]
> Quelques commentaires, mais rien de bloquant

"Quelques"

File: packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (tahini) [reply packages/transition-backend/src/models/db/__tests__/geometryUtils.db.test.ts]: If you execute the command twice, it's your problem! ;-) Why would anyone do that? But when you run it once, the run setup is such that each test is sequential with the --runInBand option.

**Comment by greenscientist** [comment]
last 2 push were the rebase. Had to redo some of the test to adapt to the new option to not calculate POIs

**Review by tahini — APPROVED** [review]
Juste valider le contenu du console.time pour éviter les conflits, sinon c'est bon

File: packages/transition-backend/src/services/accessibilityMap/TransitAccessibilityMapCalculator.ts [hunk 8]
```diff
@@ -343,13 +308,17 @@ export class TransitAccessibilityMapCalculator {
         const { isCancelled, additionalProperties } = options;
 
         try {
-            const polygons = await this.generatePolygons(attributes, result, durations, nbCalculations, {
+            //TODO Keep this time measurement for the moment, to monitor trends
+            console.time('Accessibility generatePolygons');
```
[the remark this answers is not in the document]
Comment (greenscientist) [reply packages/transition-backend/src/services/accessibilityMap/TransitAccessibilityMapCalculator.ts]: Oui, il y en a relativement souvent quand tu fais des calculs en batch.
